src/core/OIDNDenoiser.js

```diff
@@ -0,0 +1,297 @@
+// Utils to interface with the denoiser
+
+import { Denoiser } from 'denoiser';
+import { FullScreenQuad } from 'three/examples/jsm/postprocessing/Pass.js';
+import { ClampedInterpolationMaterial } from '../materials/fullscreen/ClampedInterpolationMaterial.js';
+import { DenoiserBlendMaterial } from '../materials/fullscreen/DenoiserBlendMaterial.js';
+
+import { NoBlending, WebGLRenderTarget, SRGBColorSpace } from 'three';
+
+export class OIDNDenoiser {
+
+	get weightsUrl() {
+
+		return this.denoiser.weightsUrl;
+
+	}
+
+	set weightsUrl( url ) {
+
+		this.denoiser.weightsUrl = url;
+
+	}
+
+	get quality() {
+
+		return this.denoiser.quality;
+
+	}
+
+	set quality( v ) {
+
+		this.denoiser.quality = v;
+		if ( v === 'fast' && this.hdr ) this.hdr = false;
+		//if ( v === 'balanced' && ! this.hdr ) this.hdr = true;
+
+	}
+
+	get useAux() {
+
+		return this._useAux;
+
+	}
+
+	set useAux( v ) {
+
+		this._useAux = v;
+		this.denoiser.resetInputs();
+
+	}
+
+	get cleanAux() {
+
+		return ! this.denoiser.dirtyAux;
+
+	}
+
+	set cleanAux( v ) {
+
+		this.denoiser.dirtyAux = ! v;
+
+	}
+
+	get hdr() {
+
+		return this.denoiser.hdr;
+
+	}
+
+	set hdr( v ) {
+
+		this.denoiser.hdr = v;
+		if ( v && this.quality === 'fast' ) this.quality = 'balanced';
```

Comment:
Why do we have to set the quality to "balanced" when enabling HDR?

Reply:
Went into more details in the other comment, but essentially hdr and fast quality gave the worst results I've seen. Balanced gave good results.